lib/App/DuckPAN/Perl.pm

```diff
@@ -109,54 +111,73 @@ sub duckpan_install {
 		$sp =~ s/\:\:/_/g;
 
 		# special case: check for a pinned verison number
-		my $pin_version            = $ENV{$sp};
-		my $localver               = $self->get_local_version($package);
-		my $duckpan_module_version = version->parse($module->version);
-		my $duckpan_module_url     = $self->app->duckpan . 'authors/id/' . $module->distribution->pathname;
+		my $pinned_version      = $ENV{$sp};
+		my $installed_version   = $self->get_local_version($package);
+		my $latest_version      = version->parse($module->version);
+		my $duckpan_module_url  = $self->app->duckpan . 'authors/id/' . $module->distribution->pathname;
 
-		$localver ||= 1e-6 if ($pin_version); # a silly, but true, value if missing and we need to compare with pinned.
```

Comment:
Not really.  So why can't you `duckpan upgrade` to do that?  Updating to the latest doesn't get you to prod because prod will often be pinned lower than latest.  The next time you upgrade you are mysteriously downgraded to pinned versions.  That is not good behavior.  Are you saying that you want to get the latest without updating your pins to match?